### DISCUSSION TITLE
next/head , meta tags and web scrapers (facebook)

### DISCUSSION THREAD

**Discussion by maximeMD** [vercel/next.js#22467] [Help]
Hello everybody 😄 
I have a situation here : I want to add open graph meta tags in some dynamic pages in my website, using the Head component.
Nevertheless, when i use the [facebook debugger to test my website](https://developers.facebook.com/tools/debug/) , no meta tag is set **yet**, because the scrapping made by facebook is done with the page no already rendered by javascript.

Does anyone have faced this issue yet ?

Thank you in advance for you answers !
(and sorry if it is a duplicate... )

**Comment by michrome** [comment]
Make sure your meta data is being populated using [`getStaticProps` (or `getServerSideProps`)](https://nextjs.org/docs/basic-features/pages#pre-rendering) … this will render the `<head>` server-side, before it's sent to the browser.

**Reply by maximeMD** [reply]
Thank you for your answer, but yes my data was indeed populated using getStaticProps.

The issue came from a HOC which was making a conditional rendering depending on a variable set in a React context.

Hope it will help someone passing by here :)
